topo_processor/metadata/metadata_validators/metadata_validator_stac.py

```diff
@@ -31,11 +32,15 @@ def is_applicable(self, stac_object: Union[Item, Collection]) -> bool:
         return True
 
     def validate_metadata(self, item: Item) -> None:
-        try:
+
+        with warnings.catch_warnings(record=True) as w:
```

Comment:
Should I check here if the `IterErrorsValidator` is the one used? Otherwise that won't match with the default validator (which does not raise `warnings` but `exception`). What the best way to do that? a global variable set to `True` when the custom validator is enabled? (`pystac.validation.set_validator(IterErrorsValidator())`

Reply:
Good point. Are we ever going to use the default validator? I guess it would be safest to check. How would this behave with jsonschema-rs (will we need to use that in future?).

Reply:
If we want to use the custom validator by default, something needs to be done to force it all the time (instead of setting it for `upload` as it is currently done. 

> How would this behave with jsonschema-rs (will we need to use that in future?)

Only the future will tell us :) I guess we will make a change when/if this will be needed.